src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/InlineCompletion/InlineCompletionEndPoint.cs

```diff
@@ -0,0 +1,240 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common.Extensions;
+using Microsoft.AspNetCore.Razor.LanguageServer.Extensions;
+using Microsoft.AspNetCore.Razor.LanguageServer.Formatting;
+using Microsoft.AspNetCore.Razor.LanguageServer.ProjectSystem;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.Extensions.Logging;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer;
+
+internal class InlineCompletionEndpoint : IInlineCompletionHandler
+{
+    // Usually when we need to format code, we utilize the formatting options provided
+    // by the platform. Similar to DefaultCSharpCodeActionResolver we do not have any, so use defaults.
+    private static readonly FormattingOptions s_defaultFormattingOptions = new FormattingOptions()
+    {
+        TabSize = 4,
```

Comment:
@allisonchou IIRC we had an options store on the Razor language server side that was regularly updated. Am I remembering wrong? Wondering if there's a way @dibarbet can hook into that

Reply:
if we do get the razor formatting options though, I think they still need to be passed to C# so C# formats the snippet with the right settings.  So regardless I think the best way is to just include them on the request.

Reply:
Ahhhh good point. Agreed as a good follow up!